### PR TITLE
[MM-33897][MM-33910] Revert to using original display name logic for DMs on channel switcher, removing username

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
We were ignoring the original display name code written for the channel switcher to properly format the display name for DMs. This PR restores that code, while removing the username from that code so that it doesn't display twice.

We also now show the 'Deactivated' indicator with the username, and continue to always show the username regardless of the teammate display setting.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-33897
https://mattermost.atlassian.net/browse/MM-33910

#### Screenshots
Screen Shot 2021-03-24 at 12 59 34 PM

### CODE REVIEW

**Review by asaadmahmood — APPROVED** [review]

**Review by asaadmahmood — APPROVED** [review]

File: components/suggestion/switch_channel_provider.jsx [hunk 1]
```diff
@@ -413,34 +419,16 @@ export default class SwitchChannelProvider extends Provider {
     }
 
     userWrappedChannel(user, channel) {
-        const teammateNameDisplay = getTeammateNameDisplaySetting(getState());
-        let displayName;
+        let displayName = '';
 
         // The naming format is fullname - @username (nickname) if DISPLAY_PREFER_FULL_NAME is set.
         // Otherwise, it's @username - fullname (nickname)
```
Comment (agarciamontoro) [inline components/suggestion/switch_channel_provider.jsx:425]: Does this comment sill apply?

**Review by agarciamontoro — APPROVED** [review]
I'm not sure what the requirements were, but the code looks good, though! I left a non-blocking comment below.

**Comment by jgilliam17** [comment]
Thanks @devinbinnie 
Deactivated status indicator displays as expected.
Few issues
1) I think this may be resolved by removing the status from bots that Navy is working on, just noting here that on initial load, bot name on channel switcher looks off 
Screen Shot 2021-03-25 at 3 46 17 PM

2) When First/Last name is not specified, but nickname is, I think we should probably append `(nickname)` to user name followed by status, it looks off on test server
`[username] [nickname] [status]`  maybe UX needs to confirm?
Screen Shot 2021-03-25 at 4 33 17 PM

**Comment by devinbinnie** [comment]
> 1. I think this may be resolved by removing the status from bots that Navy is working on, just noting here that on initial load, bot name on channel switcher looks off
> 
> Screen Shot 2021-03-25 at 3 46 17 PM

Didn't notice that, should we be showing an indicator of some kind there or do we just want to make sure there's some kind of spacing if there's no indicator?

> 1. When First/Last name is not specified, but nickname is, I think we should probably append `(nickname)` to user name followed by status, it looks off on test server
>    `[username] [nickname] [status]`  maybe UX needs to confirm?
> 
> Screen Shot 2021-03-25 at 4 33 17 PM

We can do this if need be, would like UX input though to confirm that's the right change. @matthewbirtch?

**Comment by jgilliam17** [comment]
1. Out of scope here. I noticed few member names in channel switcher on daily with this issue; when status fails to update/show, name and username are not spaced out correctly. I'll file a ticket for investigation. 
Screen Shot 2021-03-26 at 4 30 11 PM

**Review by nevyangelova — APPROVED** [review]
lgtm :P

**Comment by matthewbirtch** [comment]
> We can do this if need be, would like UX input though to confirm that's the right change. @matthewbirtch?

Can we remove the status for bots and also just have one name displaying? It seems odd to have the repetition of "Surveybot" and "surveybot" for example. So, it would display like this:
image

**Comment by matthewbirtch** [comment]
> 1. When First/Last name is not specified, but nickname is, I think we should probably append `(nickname)` to user name followed by status, it looks off on test server
>    `[username] [nickname] [status]`  maybe UX needs to confirm?

I don't have a problem with the way it is actually. The only thing I think we should consider is to remove the brackets from the nickname if no firstname/lastname is set

**Comment by devinbinnie** [comment]
@matthewbirtch Have a look!

**Comment by matthewbirtch** [comment]
> @matthewbirtch Have a look!

Looks good @devinbinnie. I would just add additional 4px on the left-margin of the `BOT` tag.

**Comment by jgilliam17** [comment]
Thanks @devinbinnie 
Verified bot status has been removed on Channel Switcher and Deactivated label shows as expected.
- When first/last name is not specified brackets are still present. Are we filing this separately? I may have misunderstood.
- Noticed that draft DM only shows nickname, and no other information (not reproducing on master)
Screen Shot 2021-03-31 at 8 56 19 AM

**Comment by devinbinnie** [comment]
@jgilliam17 All should be working now.

**Comment by devinbinnie** [comment]
@jgilliam17 I've updated the branch so that issue shouldn't be affecting us anymore.

**Comment by jgilliam17** [comment]
Discussed the remaining issue concerning `draft` DMs with Devin before long weekend. Just adding a note here so I don't forget
- Draft DMs were not displaying complete user information, and became `invisible` on channel switcher after I had rebuilt the test server.
Screen Shot 2021-04-06 at 10 37 19 PM

**Comment by devinbinnie** [comment]
@jgilliam17 Should be fixed now.

**Comment by jgilliam17** [comment]
Thanks @devinbinnie, apologize for delay 
Draft DMs are displaying as expected.
One small spacing issue, dash in `- Deactivated` is too close to status icon
Screen Shot 2021-04-12 at 6 46 11 PM

**Comment by devinbinnie** [comment]
> Thanks @devinbinnie, apologize for delay
> Draft DMs are displaying as expected.
> One small spacing issue, dash in `- Deactivated` is too close to status icon (edit: when name and nickname are not set)
> Screen Shot 2021-04-12 at 6 46 11 PM

@jgilliam17 I've moved the `- Deactivated` bit next to the username which should look better.

**Review by jgilliam17 — APPROVED** [review]
Thank you @devinbinnie 
Tested, looks good to merge.
- Verified nickname and deactivated label for DMs on channel switcher.

**Pull request merged**